### PR TITLE
Homework 19

### PR DESCRIPTION
**Pull request by eolsen3**
I keep getting the message : Unable to find element with locator By.cssSelector: input[type='email'].
So now Homework 18 won't run, and neither will Homework 19. I've tried using other selectors, to no avail. What's the issue?

### CODE REVIEW

File: src/test/resources/Homework18.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
```
Comment (rajniss21) [inline src/test/resources/Homework18.xml:3]: HomeWork18 missing "BaseUrl" parameter:
`<parameter name = "BaseUrl" value="https://qa.koel.app/"/>`

File: src/test/java/Homework19.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        provideEmail("erika.olsen@testpro.io");
+        providePassword("te$t$tudent");
+        clickSubmit();
+        openPlaylist();
+        clickDeletePlaylistBtn();
+        Assert.assertTrue(getDeletedPlaylistMsg().contains(deletedPlayListMsg));
+    }
+
+   public void openPlaylist() {
+       WebElement clickTestProPlaylist = driver.findElement(By.cssSelector("[href='#!/playlist/64251']"));
```
Comment (rajniss21) [inline src/test/java/Homework19.java]: Your defined CSS selector was unable to find the WebElement.
 `WebElement clickTestProPlaylist = driver.findElement(By.cssSelector("section[id='playlists']>ul>li:nth-child(3)"));`

**Comment by rajniss21** [comment]
Hello,
I have checked your homework 18 & 19 But you have got this error while compiling the code. Please review your code, fix it, and complete your homework.

- In HomeWork19 you got this assertion error. Please review it.
- HomeWork18 failed was of missing BaseURL.

Best wishes from the HW team (Rajnish, rajnish@testpro.io)!

Screenshot_43
Screenshot_42

File: src/test/java/Homework19.java (hunk 2 above)
Comment (eolsen3) [reply src/test/java/Homework19.java]: Can you explain to me how you got i:nth-child(3)"))?
I can see up to that point.
Screenshot 2023-07-18 at 10 29 27 AM

**Comment by eolsen3** [comment]
I updated Homework 18, and Homework 19 I keep getting an error that the boolean expects to return true, but it's false. I can watch it go through the test successfully. It's not saying it can't find the locator, I don't know what the problem is.
I also tried div.success.show with no success.

**Comment by Berichicko** [comment]
> I updated Homework 18, and Homework 19 I keep getting an error that the boolean expects to return true, but it's false. I can watch it go through the test successfully. It's not saying it can't find the locator, I don't know what the problem is. I also tried div.success.show with no success.

You got this problem, please correct it. 

image

**HINT**

 `WebElement successMessage = driver.findElement(By.cssSelector("div.success.show"));`
 
 
image


image

**Comment by Berichicko** [comment]
You got this problem, please correct it. 


image

_I will try to help you how easy it is to find locators:_

**1) These steps describe the process of opening the developer console in the Google Chrome browser and making changes to the styles of a popover that appears when hovering over a specific element on a web page.**

For example

 `By.cssSelector("div.success.show")`

image

image

Browse to the desired page
1.	Open the dev console - F12 on Windows/Linux or option + ⌘ + J on macOS
2.	Select the Sources tab in chrome inspector
3.	In the web browser window, hover over the desired element to initiate the popover
4.	Hit F8 on Windows/Linux (or fn + F8 on macOS or  CTRL+\) while the popover is showing. If you have clicked anywhere on the actual page F8 will do nothing. Your last click needs to be somewhere in the inspector, like the sources tab
5.	Go to the Elements tab in inspector
6.	Find your popover (it will be nested in the trigger element's HTML)
7.	Have fun modifying the CSS
8.	https://www.howtogeek.com/663950/what-your-function-keys-do-in-chrome-devtools/



**2) You can use this extension for easy locator search.**

 Ranorex Selocity

Ranorex Selocity is a free extension for Chrome DevTools. It allows you to create unique selectors through a smart selector generator. Feedback is created, allowing you to locate your web page's elements in the most secure manner. https://www.ranorex.com/selocity/browser-extension/

For example 

image

**3) Test**

image


```
    @Test
    public void deletePlaylist () throws InterruptedException {
        String deletedPlaylistMessage = "Deleted playlist \"" + "Test Pro Playlist.\"";

        provideEmail("erika.olsen@testpro.io");
        providePassword("te$t$tudent");
        clickSubmit();
        openPlaylist();
        clickDeletePlaylistBtn();
        Assert.assertTrue(getDeletedPlaylistMsg().contains(deletedPlaylistMessage));
    }
```
`String deletedPlaylistMessage = "Deleted playlist \"" + "Test Pro Playlist.\"";`

In Java, strings can be constructed by enclosing characters in double quotes (" "). The backslash (\) before the second double quote (") is an escape character used to include a double quote as part of the string. Therefore, the resulting string includes the text Deleted playlist followed by the name of the playlist, which in this case is Test Pro Playlist.. The backslashes help represent the quotes as part of the string without ending the string prematurely.

**The value of this string is:**

**Deleted playlist "Test Pro Playlist."**

image

image

image

**Comment by Berichicko** [comment]
Hello,

I have checked your homework 19, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

804deb64dc406a20a75da05d91a1a69e

**Excellent! Perfect! Brilliant! Fantastic! Fantastic job! Magnificent! Terrific! Marvellous. Superb! Wow! Great stuff! You’re doing great! Keep it up! Very fine. Hooray for you! That’s better than ever! Super work!** 

image

**Pull request closed without merging**